jsonstar/encoder.py

```diff
@@ -20,26 +20,58 @@ def __setitem__(self, type_, function):
                 self.move_to_end(base)
 
 
-class JSONEncoderStar(stdlib_json.JSONEncoder):
+class EncoderMeta(type):
+    def __new__(mcs, name, bases, namespace):
+        if "_default_functional_encoders" not in namespace:
+            namespace["_default_functional_encoders"] = []
+
+        if "_default_typed_encoders" not in namespace:
+            namespace["_default_typed_encoders"] = TypedEncoderRegistry()
+
+        return super().__new__(mcs, name, bases, namespace)
+
+
+class JSONEncoderStar(stdlib_json.JSONEncoder, metaclass=EncoderMeta):
     class FUNCTIONAL:
         """Sentinel type to register a functional encoder."""
 
-    default_typed_encoders = TypedEncoderRegistry(DEFAULT_TYPED_ENCODERS)
-    default_functional_encoders = DEFAULT_FUNCTIONAL_ENCODERS
+    _default_typed_encoders = TypedEncoderRegistry(DEFAULT_TYPED_ENCODERS)
+    _default_functional_encoders = DEFAULT_FUNCTIONAL_ENCODERS
 
     def __init__(self, *args, functional_encoders=(), typed_encoders: dict[type, callable] = NULL_DICT, **kwargs):
         super().__init__(*args, **kwargs)
 
         self._typed_encoders = TypedEncoderRegistry(typed_encoders)
         self._functional_encoders = [*functional_encoders]
 
+    @classmethod
+    def default_functional_encoders(cls):
+        if cls is JSONEncoderStar:
+            return cls._default_functional_encoders
+        else:
+            return cls._default_functional_encoders + cls.__base__.default_functional_encoders()
+
+    @classmethod
+    def default_typed_encoders(cls):
+        if cls is JSONEncoderStar:
+            return cls._default_typed_encoders
+        else:
+            a = cls._default_typed_encoders
+            b = cls.__base__.default_typed_encoders()
+            c = ChainMap(a, b)
+            d = TypedEncoderRegistry(c)
+            return d
+            # return TypedEncoderRegistry(ChainMap(cls.__base__.default_typed_encoders(), cls._default_typed_encoders))
```

Comment:
Did you mean to leave it like this or was this for debugging and L64 is what needs to stay?

Reply:
It was for debugging. I will fix it. Thank you

Reply:
Fixed